app/controllers/application_controller.rb

```diff
@@ -5,6 +5,130 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "1234hah"
+  end
+  
+  get '/' do
+  	erb :'/index'
+  end
+
+  get '/signup' do
+    if !!session[:user_id]
+    	redirect '/tweets'
+    else
+    	erb :'/users/create_user'
+    end
+  end
+
+  post '/signup' do
+  	if params.any?{|key, value| value.empty?}
```

Comment:
I saw you used ActiveRecord validations in your Tweet model. You could have also use them here as well. 

[ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.